tests/Feature/ProductTest.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+use App\Models\Brand;
+use App\Models\Color;
+use App\Models\Product;
+use App\Models\Size;
+
+it('user can add a new product', function () {
+
+    userLogin();
+
+    $productData = [
+        'name' => 'PDU',
+        'brand_id' => Brand::factory()->create()->id,
+        'size_id' => Size::factory()->create()->id,
+        'color_id' => Color::factory()->create()->id,
+        'status' => true,
+    ];
+
+    $this->post(route('product.store'), $productData)
+        ->assertStatus(302)
+        ->assertRedirect(route('product.index'));
+
+    $this->assertDatabaseHas('products', $productData);
+});
+
+it('user can update a product', function () {
+
```

Comment:
add test cases that required filed missing in the request than getting validation error.